### DISCUSSION TITLE
WMS/WMTS

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#2540] [Q&A]
Hello,

I work on a app which use WMS/WMTS layers. I have any problems about that. When I declare a new layer WMS, I have this error "Could not load image because of The source image could not be decoded.. Please make sure to use a supported image type such as PNG or JPEG. Note that SVGs are not supported".

I tried with several sources, it seems that this problem concerns all WMS/WMTS declared in GetCapabilities. I have a IGN layer declared in GetTile, and I didn't have this problem.

There is a solution about this problem ?

Thank you very much for your contributions,

Sylvain

**Comment by IvanSanchez** [comment]
I'm gonna bet this is one of those "WMS server is returning an error in XML format" scenarios. Use your browser's developer tools to see what's the actual response from the WM(T)S server.

**Reply by sylvain258** [reply]
Hello Ivan, thank you four your return. For your information, these WMS works correctly on qgis. So, urls are ok. The problem come from Maplibre. In my console, the request wms returns status 200 but with message "Echec du chargement des données de la réponse : Request content was evicted from inspector cache".  I don't know that it means.... The problem doesn't come from of the server localhost. I tried on another server, and I have the same problem. 

What do you think?
Thank you for your help
Sylvain

**Comment by gtnbssn** [comment]
I am experiencing something similar here.

This url works well in QGIS, GDAL, and openlayers: https://libmaps.nus.edu.sg/services/1846/wmts

However with MapLibre i get the following:

`Error: Could not load image because of An attempt was made to use an object that is not, or is no longer, usable. Please make sure to use a supported image type such as PNG or JPEG. Note that SVGs are not supported.`

Looking a the network tab, I see MapLibre is requesting the same url over and over, without adjusting parameters, and therefore always getting the GetCapabilities XML. So it is indeed wrongly receiving xml, but I had assumed MapLibre would be doing the job of forming the request for each tile!

Do I need to add more parameters in the url?

Here is a codepen that reproduces the issue:

https://codepen.io/gbn000/pen/xxBxdQN

For what it's worth, the WMTS is provided by [titiler](https://github.com/developmentseed/titiler)

**Reply by HarelM** [reply]
Yes, you'll need to add more parameters, maplibre doesn't read the capablilities file, it only works with the tiles address.
Here's an example:
https://maplibre.org/maplibre-gl-js/docs/examples/wms/

**Reply by gtnbssn** [reply]
Thanks!

In this case I think we will just be using TMS then. I think the main interest in using WMTS would be if it would respect the bounding box of the layer by fetching it from the capabilities, but here in both cases I will be requesting tiles that are outside of the extent.

(OpenLayers seems to be the only library doing this at the moment, and it currently has a bug so that isn't even the case now for us: https://github.com/openlayers/openlayers/issues/15363.)
